tools/xtce_generation/telemetry_generator.py

```diff
@@ -1,16 +1,20 @@
 import xml.etree.ElementTree as Et
 from datetime import datetime
 
+import pandas as pd
+
 from tools.xtce_generation.ccsds_header_xtce_generator import CCSDSParameters
 
 
 class TelemetryGenerator:
     def __init__(self, packet_name, path_to_excel_file, apid, sci_byte=None, pkt=None):
         self.packet_name = packet_name
-        self.path_to_excel_file = path_to_excel_file
         self.apid = apid
         self.sci_byte = sci_byte
-        self.pkt = pkt
+
+        # Read excel sheet
+        xls = pd.ExcelFile(path_to_excel_file)
+        self.pkt = xls.parse(packet_name)
```

Comment:
Can you give an example? 

Removing column should not effect it. We mainly use these column in generator - `lengthInBits`, `dataType`, `mnemonic`, `longDescription`, `shortDescription`. If those column contains, it should still work.

Reply:
The GLOWS sheet has a column which sums up all the bits - presumably as a check for the size of the packet. If I wanted to remove that programmatically, I wouldn't be able to, because the excel file is parsed inside this class instead of the class having the packet passed in as a dataframe. 

```
xls = pd.ExcelFile(excel_file)
    packet_df = xls.parse(packet_name)

    # Clean up packet
    packet_df.drop(
        labels=packet_df.columns[packet_df.columns.str.contains("Unnamed")],
        axis=1,
        inplace=True,
    )
    packet_df.dropna(subset="packetName", inplace=True)

    generator = GlowsTelemetryGenerator(packet_name, excel_file, apid, pkt=packet_df)
```

Reply:
it's hard to explain, and barely matters 😅 but this bit of code just chops off the unnamed columns and cleans up the sheet.